Show Variations screen on tapping "Add price to your variations..." warning row

selanthiraiyan opened the pull request:
Closes #4849

## Description
“Add price to your variations to make them visible…” warning in `ProductFormViewController` was not tappable and an issue was reported to make it tappable and also launch the Variations screen on tap. 
This change includes the code to make the warning row actionable and also launch the `ProductVariationsViewController` on tap.

## Testing
You should add a Product with a variant, which doesn't have a price set. (Please refer to the attached recording.)

1. Go to the `Products` tab.
2. Tap on the product which has a variant and no price set.
3. In the tableview, you should see the "Add price to your variations to make them visible on your store" warning, just below the `Variations` row.
4. Tap on "Add price to your variations to make them visible on your store" warning and observe that `Variations` screen is shown.
5. Navigating back and tapping on `Variations` row should also end up showing the `Variations` screen.


https://user-images.githubusercontent.com/524475/135368677-cc101da8-4c29-43dc-9270-3ba702a9d2c2.mov

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Actionable `.noPriceWarning` row
- Make `noPriceWarning` actionable and show variations screen on tap.
Closes #4849

selanthiraiyan commented on WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift:
```diff
@@ -359,23 +359,22 @@ final class ProductFormViewController<ViewModel: ProductFormViewModelProtocol>:
-                    viewModel.updateProductVariations(from: updatedProduct)
+                showVariations()
+            case .noPriceWarning(let viewModel):
+                ServiceLocator.analytics.track(.productDetailViewVariationsTapped)
```
Should we use a different event to track this?

Ecarrion replied:
I think we can leave this, and treat it as it was a big cell.

Commit:
Merge branch 'develop' into issue/4849-actionable-add-price-to-variations-row

joshbetz commented:
Thanks for the PR @selanthiraiyan! We're taking a look.

Ecarrion commented on WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift:
```diff
@@ -359,23 +359,22 @@ final class ProductFormViewController<ViewModel: ProductFormViewModelProtocol>:
+                showVariations()
+            case .noPriceWarning(let viewModel):
+                ServiceLocator.analytics.track(.productDetailViewVariationsTapped)
+                guard viewModel.isActionable else {
+                    return
```
I think this guard should go before the analytics, or we will be login taps on un-tappable rows, like on the product variation view.


https://user-images.githubusercontent.com/562080/135634944-e1d65664-8fba-472f-8288-b2bff10b4f2e.mov

selanthiraiyan replied:
Thanks for reviewing @Ecarrion ! ❤️ 
You are right. I left it this way, because we were previously logging the event before the guard. I will fix it.

Ecarrion reviewed (changes requested):
Hi @selanthiraiyan,

Thanks for giving this a go! 🎉 

I just left you a small suggestion and we should be good to go!

Commit:
Variations tap event
- Log events only if the rows are actionable.

Ecarrion reviewed (changes requested):
@selanthiraiyan I'm seeing that the analytics event is always fired, I guess that is because you are setting `WarningViewModel.actionable` to be always true.

We should modify that to be false for product variations.

Commit:
noPriceWarningRow in ProductVariations
- Set isActionable as false when presenting the warning in ProductVariations screen.

selanthiraiyan commented:
> @selanthiraiyan I'm seeing that the analytics event is always fired, I guess that is because you are setting `WarningViewModel.actionable` to be always true.
> 
> We should modify that to be false for product variations.

I understand and thanks for your suggestion on changing `isActionable` to false for product variations. I have pushed those changes.

Commit:
Merge branch 'develop' into issue/4849-actionable-add-price-to-variations-row

Ecarrion reviewed (approved):
This looks great now! @selanthiraiyan 
I'll merge it as soon as CI finishes! 🎉
